tests/pipelines/test_pipelines_summarization.py

```diff
@@ -67,8 +67,8 @@ def run_pipeline_test(self, summarizer, _):
             # the embedding layer.
             if not (
                 isinstance(model, TFPreTrainedModel)
-                and get_gpu_count() > 0
                 and len(summarizer.model.trainable_weights) > 0
+                and "GPU" in summarizer.model.trainable_weights[0].device
```

Comment:
Quick note to any reviewers so they can make sense of this one: The problem is that `get_gpu_count()` is supposed to be framework-neutral, but actually checks the frameworks in order using `is_torch_available()` etc. and returns the GPU count for the first framework that matches.

This is very risky for TensorFlow, because TF environments will often have Torch as well, and if Torch is present then the *Torch* GPU count is returned instead, which may be 0 even if TF is running on GPU. 

I just refactored the check to not use that function.